pcdsutils/log.py

```diff
@@ -243,18 +247,31 @@ def configure_pcds_logging(
     return handler
 
 
-def validate_log_level(level) -> int:
+def validate_log_level(level: typing.Union[str, int]) -> int:
     '''
-    Return a int for level comparison
+    Return a logging level integer for level comparison.
+    Parameters
+    ----------
+    level : str or int
+        The logging level string or integer value.
+    Returns
+    -------
+    log_level : int
+        The integral log level.
+    Raises
+    ------
+    ValueError
+        If the logging level is invalid.
     '''
     if isinstance(level, int):
-        levelno = level
-    elif isinstance(level, str):
+        return level
+    if isinstance(level, str):
         levelno = logging.getLevelName(level)
 
-    if isinstance(levelno, int):
-        return levelno
-    raise ValueError("Log level is invalid")
+    if not isinstance(levelno, int):
+        raise ValueError("Invalid logging level (use e.g., DEBUG or 6)")
```

Comment:
```suggestion
        raise ValueError(f"Invalid logging level {levelno!r} (use e.g., DEBUG or 6)")
```

Reply:
I'll do this, but with some adjustments to avoid breaking the line length limit

Reply:
RE: Tyler's comment, I had to google a bit to understand the context: https://www.python.org/dev/peps/pep-0498/#s-r-and-a-are-redundant
I think maybe we should switch to `repr()` for clarity but I don't have a strong feeling about it

Reply:
`!r` is something I'm used to and familiar with, but don't really mind either way. `repr()` being longer is probably the only count against it.

Reply:
I will be lazy and leave it as !r then :shrug: